### PR TITLE
Create heap dump on OOME on nightly build

### PR DESCRIPTION
**Pull request by Holmistr**
This should help me investigate, why the JoinStressTest is failing in the lab with OOME.

### CODE REVIEW

**Comment by Holmistr** [comment]
@Donnerbart I decided to go with the easier way and remote the "Delete workspace after build" temporarly, so I can easily get it tomorrow.

File: pom.xml [hunk 1]
```diff
@@ -982,6 +982,7 @@
                             <useFile>false</useFile>
                             <argLine>
                                 -Xms128m -Xmx1G -XX:MaxPermSize=1024M
+                                -XX:+HeapDumpOnOutOfMemoryError
```
Comment (kwart) [inline pom.xml]: I prefer aligning additional JVM arguments with master branch and using the `${extraVmArgs}` across the file.

Comment (Donnerbart) [reply pom.xml]: That sounds like a good idea, didn't think of it. With that it should be possible to configure the `-XX:+HeapDumpOnOutOfMemoryError` via the build configuration. Good one!

**Comment by Holmistr** [comment]
@Donnerbart @kwart Addressed your comment :) Nice one!

**Comment by Holmistr** [comment]
Damn, I need to fix it one more time :D

**Comment by Holmistr** [comment]
@Donnerbart @kwart Ok, now it should be final :)

**Pull request merged**